### PR TITLE
DT-2129: validate email address supplied for forgotten username

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 plugins {
   id("uk.gov.justice.hmpps.gradle-spring-boot") version "3.3.0"
-  kotlin("plugin.spring") version "1.5.10"
-  kotlin("plugin.jpa") version "1.5.10"
-  id("org.jlleitschuh.gradle.ktlint") version "10.0.0"
+  kotlin("plugin.spring") version "1.5.20"
```
Comment (petergphillips) [inline build.gradle.kts:3]: shouldn't really upgrade this unless actually upgrading kotlin - could try `3.3.1-beta`

File: build.gradle.kts [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+  kotlin("plugin.jpa") version "1.5.20"
+  id("org.jlleitschuh.gradle.ktlint") version "10.1.0"
```
Comment (petergphillips) [inline build.gradle.kts:5]: I think this is in the plugin these days too?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/ForgottenUsernameController.kt [hunk 3]
```diff
@@ -60,6 +65,14 @@ class ForgottenUsernameController(
         }
       }
       modelAndView
+    } catch (e: ValidEmailException) {
+      log.error("Failed to send forgotten username email due to", e)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/ForgottenUsernameController.kt:69]: this log message doesn't look right

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/ForgottenUsernameController.kt [hunk 4]
```diff
@@ -60,6 +65,14 @@ class ForgottenUsernameController(
...
+      telemetryClient.trackEvent(
+        "ForgottenUsernameRequestFailure",
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/ForgottenUsernameController.kt:71]: don't we prefix all our events with `Auth`?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/ForgottenUsernameController.kt [hunk 5]
```diff
@@ -60,10 +65,18 @@ class ForgottenUsernameController(
         }
       }
       modelAndView
+    } catch (e: ValidEmailException) {
+      log.error("Forgotten username email failed validation")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/ForgottenUsernameController.kt:69]: bit naff this is an `error` - it is caused by the client entering invalid data so really could be an `info`, or don't need it at all...

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/ForgottenUsernameController.kt (hunk 5 above)
Comment (SimonMitchellMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/ForgottenUsernameController.kt:69]: will remove as don't think we need it really

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**